compiler/functions.go

```diff
@@ -44,6 +45,44 @@ func (fc *funcContext) nestedFunctionContext(info *analysis.FuncInfo, sig *types
 		c.allVars[k] = v
 	}
 
+	// Synthesize an identifier by which the function may reference itself. Since
+	// it appears in the stack trace, it's useful to include the receiver type in
+	// it. Note that we are using a middle dot unicode character to delimit
+	// between receiver and the function name, since it's a valid identifier
+	// character in JS.
+	funcRef := o.Name()
+	if typeName := c.sigTypes.RecvTypeName(); typeName != "" {
+		funcRef = typeName + "·" + funcRef
```

Comment:
I wonder if it might be a good idea to define a constant for `"·"`, to both make it more visually distinct (this just looks like a period, without additional textual context or very close visual inspection), and to make it easier to reproduce.